### PR TITLE
Fix migration issue.

### PR DESCRIPTION
**Pull request by johnsonw**
Signed-off-by: Will Johnson <william.c.johnson@intel.com>

### CODE REVIEW

File: chroma-manager/chroma_core/migrations/0023_auto__add_unconfigurersyslogjob__add_getcorosyncstatejob__add_pacemake.py [hunk 1]
```diff
@@ -4,6 +4,14 @@
 from south.v2 import SchemaMigration
 from django.db import models
 
+def create_table_dict (model_name, state, host_id):
+    return { k: v for k, v in (
+        ('state_modified_at', datetime.datetime.now()),
+        ('state', state),
+        ('immutable_state', 'f'),
+        ('host_id', host_id),
+        ('corosync_reported_up', 'False' if model_name == 'corosyncconfiguration' else None)
```
Comment (jgrund) [inline chroma-manager/chroma_core/migrations/0023_auto__add_unconfigurersyslogjob__add_getcorosyncstatejob__add_pacemake.py]: Everything else in this is static. 

Why not just treat this as a constructor and return a dict?

File: chroma-manager/chroma_core/migrations/0023_auto__add_unconfigurersyslogjob__add_getcorosyncstatejob__add_pacemake.py [hunk 2]
```diff
@@ -232,15 +240,17 @@ def forwards(self, orm):
...
             ct, _ = orm['contenttypes.ContentType'].objects.get_or_create(model=model_name.lower(),
                                                                           app_label=app_name,
                                                                           defaults=dict(name=model_name))
 
-            db.execute("insert into chroma_core_%s (state_modified_at, state, immutable_state, content_type_id, host_id) "
-                       "values ('%s', '%s', 'f', %s, %s)" % (model_name, datetime.datetime.now(), state, ct.id, host_id))
-
+            table_dict['content_type_id'] = ct.id
+            fields = ", ".join(map(str, table_dict.keys()))
+            values = ", ".join(map(str, map(lambda x: "'%s'" % x, table_dict.values())))
```
Comment (jgrund) [inline chroma-manager/chroma_core/migrations/0023_auto__add_unconfigurersyslogjob__add_getcorosyncstatejob__add_pacemake.py]: couldn't you just get the items here and then unzip them?

File: chroma-manager/chroma_core/migrations/0023_auto__add_unconfigurersyslogjob__add_getcorosyncstatejob__add_pacemake.py (hunk 1 above)
Comment (johnsonw) [reply chroma-manager/chroma_core/migrations/0023_auto__add_unconfigurersyslogjob__add_getcorosyncstatejob__add_pacemake.py]: I'm not sure I understand the benefit. I'm not opposed to it, can we discuss?

File: chroma-manager/chroma_core/migrations/0023_auto__add_unconfigurersyslogjob__add_getcorosyncstatejob__add_pacemake.py (hunk 2 above)
Comment (johnsonw) [reply chroma-manager/chroma_core/migrations/0023_auto__add_unconfigurersyslogjob__add_getcorosyncstatejob__add_pacemake.py]: hmm, maybe. This takes something like:
```
{
   'state_modified_at': datetime.datetime.now(),
   'state', 'managed',
   'immutable_state': 'f',
   'host_id': '9'
}
```
and gives me:
```
fields = "state_modified_at, state, immutable_state, host_id"
values = "'2018-06-06', 'managed', 'f', '9'"
```
Perhaps I'm misunderstanding, but doesn't zip take two arrays and create a new list of tuples?

**Comment by johnsonw** [comment]
image

**Comment by johnsonw** [comment]
image

**Pull request merged**